src/strategy.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MinimalRenamer} from './minimal-renamer';
+import {type SkipPredicate} from './skip';
+
+/**
+ * Renaming function.
+ * A renaming function must satisfy the following:
+ * - a renaming function always produces the same output for a specific input
+ * - a renaming function does not produce anything that is in the user-specified
```

Comment:
Wait, is the skiplist supposed to also limit what kind of output this produces? I thought it was only supposed to limit which CSS names to process 🤔